### PR TITLE
search: filematch specific columns in exhaustive CSV output

### PR DESCRIPTION
**Pull request by keegancsmith**
This updates the format of our downloads to be more in line with the the CSVs users can already download from the web app. See the comments we added for different decisions we made. Originally we wanted to have the same columns, but we don't believe the columns in our export CSV are that useful. So we are taking this opportunity to experiment with something that may be more useful.

It also updates the log format as follows:
- repo name instead of repo id
- more descriptive column names

Fixes https://github.com/sourcegraph/sourcegraph/issues/56821

Test Plan: updated test

### CODE REVIEW

**Comment by keegancsmith** [comment]
cc @stefanhengl

File: internal/search/result/file.go [hunk 1]
```diff
@@ -261,6 +275,73 @@ func (hs ChunkMatches) AsLineMatches() []*LineMatch {
...
+		)
+	}
+
+	return json.Marshal(lineRange)
+}
+
+// FormatCSV returns a string representation of ChunkMatches which matches the
+// download format of the search results page. See searchResultsExport.ts for
+// comparison.
+func (hs ChunkMatches) FormatCSV() string {
```
Comment (stefanhengl) [inline internal/search/result/file.go:324]: Remove before merging

File: internal/search/exhaustive/service/matchcsv.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		return err
+	} else if ok {
+		if err := w.w.WriteHeader(
+			"Match type",
+			"Repository",
+			"Revision",
+			"Repository external URL",
+			"File path",
+			"File URL",
+			"Line ranges",
```
Comment (stefanhengl) [inline internal/search/exhaustive/service/matchcsv.go:50]: I am not a fan of the name, suggestions welcome.

File: cmd/frontend/internal/search/httpapi/export.go [hunk 3]
```diff
@@ -12,6 +13,11 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+// search-jobs_<job-id>_2020-07-01_150405
+func filename(jobID int) string {
+	return fmt.Sprintf("search-jobs_%d_%s.csv", jobID, time.Now().Format("2006-01-02_150405"))
```
Comment (keegancsmith) [inline cmd/frontend/internal/search/httpapi/export.go:18]: call sites already append an extension. Maybe rename function to indicate it's filename prefix?
```suggestion
	return fmt.Sprintf("search-jobs_%d_%s", jobID, time.Now().Format("2006-01-02_150405"))
```

File: internal/search/result/file.go [hunk 4]
```diff
@@ -23,11 +26,22 @@ type File struct {
...
+
+func (f *File) url(atCommit bool) *url.URL {
 	var urlPath strings.Builder
 	urlPath.Grow(len("/@/-/blob/") + len(f.Repo.Name) + len(f.Path) + 20)
 	urlPath.WriteRune('/')
 	urlPath.WriteString(string(f.Repo.Name))
-	if f.InputRev != nil && len(*f.InputRev) > 0 {
+	if atCommit {
+		urlPath.WriteRune('@')
+		urlPath.WriteString(string(f.CommitID))
```
Comment (keegancsmith) [inline internal/search/result/file.go:43]: is it worth adding this field? I know we have it in the export CSV, but it feels like data that a person could easily construct from the columns?

File: internal/search/result/file.go [hunk 5]
```diff
@@ -261,6 +275,73 @@ func (hs ChunkMatches) AsLineMatches() []*LineMatch {
...
+			}
+		}
+		builder.WriteString("]") // end of lines
+		builder.WriteString("]") // end of chunk
+		if i < len(hs)-1 {
+			builder.WriteString("; ")
+		}
+	}
+	return builder.String()
+}
```
Comment (keegancsmith) [inline internal/search/result/file.go:343]: surprising to me that this logic was added here. it feels search jobs specific and is only accessing exported fields. This sort of logic should live in search jobs.

**Review by keegancsmith — COMMENTED** [review]
@stefanhengl will take a deeper look soon.

File: internal/search/result/file.go (hunk 4 above)
Comment (stefanhengl) [reply internal/search/result/file.go:43]: I don't know. It's a tradeoff between convenience and size of the download. While testing it was pretty nice to just click on the link to see if the results make sense. 

I think the repo URL is much more debatable.

Comment (stefanhengl) [reply internal/search/result/file.go:43]: The minimal set of fields is IMO
- Repository
- Revision
- File path
- Line range (or better name)

Everything else is just convenience.

File: internal/search/exhaustive/service/search_test.go [hunk 6]
```diff
@@ -14,62 +14,9 @@ import (
 
 	"github.com/sourcegraph/sourcegraph/internal/actor"
 	"github.com/sourcegraph/sourcegraph/internal/search/client"
-	"github.com/sourcegraph/sourcegraph/internal/search/exhaustive/types"
 	"github.com/sourcegraph/sourcegraph/internal/uploadstore/mocks"
 )
 
-func TestBackendFake(t *testing.T) {
```
Comment (stefanhengl) [inline internal/search/exhaustive/service/search_test.go:21]: Move this around because autogold fails if the logic is not all in one file.

**Review by stefanhengl — APPROVED** [review]
Let's merge!

**Comment by keegancsmith** [comment]
@stefanhengl I just updated the description (PTAL), but I will take one more look at the diff then merge.

**Review by keegancsmith — COMMENTED** [review]
LGTM

**Pull request merged**